django/db/models/sql/subqueries.py

```diff
@@ -83,13 +84,42 @@ def add_update_values(self, values):
         query. This is the entry point for the public update() method on
         querysets.
         """
-        values_seq = []
+        field_dict = dict()
         for name, val in values.items():
+            name, *transforms = name.split(LOOKUP_SEP)
             field = self.get_meta().get_field(name)
             direct = (
                 not (field.auto_created and not field.concrete) or not field.concrete
             )
             model = field.model._meta.concrete_model
+            lhs = name
+
+            # Support using transforms in update()
+            if transforms:
+                for transform in transforms:
+                    transform_class = field.get_transform(transform)
+                    if transform_class:
+                        lhs = transform_class(lhs)
+                    else:
+                        raise FieldError(
+                            f"{transform} is not a valid Transform "
+                            f"on {field.__class__.__name__}"
+                        )
```

Comment:
This logic seems similar to the one that already exists in `Query.try_transform()`. We've discussed this before and encountered a few issues in doing so, but I still wonder if it's possible to reuse that method. I'll do some experiments later.